gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskTest.groovy

```diff
@@ -1,10 +1,159 @@
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+            }
+        '''.stripIndent().replace("{{mavenRepo}}", mavenRepo.getAbsolutePath())
+
+        when:
+        BuildResult buildResult = run(':testCreateManifest').buildAndFail()
+
+        then:
+        buildResult.task(':testCreateManifest').outcome == TaskOutcome.FAILED
+        buildResult.output.contains("The following products are recommended as dependencies but do not appear in the " +
+                "product dependencies or product dependencies ignored list: [group:name2, group:name]")
+    }
+
+    def 'Can ignore recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///{{mavenRepo}}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+                ignoredProductIds = [
+                    new com.palantir.gradle.dist.ProductId("group:name"), 
+                    new com.palantir.gradle.dist.ProductId("group:name2")
+                ]
+            }
+        '''.stripIndent().replace("{{mavenRepo}}", mavenRepo.getAbsolutePath())
+
+        when:
+        BuildResult buildResult = run(':testCreateManifest').build()
+
+        then:
+        buildResult.task(':testCreateManifest').outcome == TaskOutcome.SUCCESS
+    }
+
+    def "Can set product dependencies from recommended product dependencies"() {
+        setup:
+        generateDependencies()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///{{mavenRepo}}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = [
+                    new com.palantir.gradle.dist.ProductDependency("group", "name"),
```

Comment:
1. could we use strings here instead as well?
2. could this use a dependency configuration: `slsProduct` or something similar? (so it would use `runtime` + `slsProduct`, or `slsProduct` extends runtime)